src/ORM/ManyManyList.php

```diff
@@ -433,6 +433,58 @@ public function removeAll()
         }
     }
 
+    /**
+     * Set the extra field data for a single row of the relationship join
+     * table, given the known child ID.
+     *
+     * @param int $itemID The ID of the child for the relationship
+     * @param array $data The data to set, with field names as keys and values as values
+     * @throws BadMethodCallException if the foreign ID has not been set
```

Comment:
I think this needs to be updated to `InvalidArgumentException`?

Reply:
Yes, true. That was from my failed `SQLUpdate` implementation. I've updated the exception class and description here.